sycl/include/CL/sycl/detail/queue_impl.hpp

```diff
@@ -53,6 +54,7 @@ class queue_impl {
   }
 
   ~queue_impl() {
+    throw_asynchronous();
```

Comment:
This doesn't look as a refactoring. I think it is better to move this change at least to a separate commit

Reply:
Done.